Merge new microraiden documentation version

seichris opened the pull request:
- Restructured documentation
- New setup tutorial
- Text & wording fixes

ezdac committed:
integrate README, change path in config

ezdac committed:
Add version parsing, add extensions

ezdac committed:
Add first directives for click and autodoc

ezdac committed:
Integrate README content in docs

ezdac committed:
Change docstring to google style

ezdac committed:
Include apidoc files in repository

ezdac committed:
Remove m2m client doc

ezdac committed:
Add requirements.txt for building the docs

ezdac committed:
Restructure docs

ezdac committed:
Fix sphinx compile errors

ezdac committed:
Update doc requirements

ezdac committed:
Fix requirements for standalone usage

ezdac committed:
Add blockchain setup tutorial

ezdac committed:
Change wording

ezdac committed:
Add diagram to index file

ezdac committed:
Move quickstart to tutorials folder (not indexed)

ezdac committed:
Improve tutorials

ezdac committed:
Fix Version file location

ezdac committed:
Hotfix: exlude autodoc to avoid importing of source

ezdac committed:
Fix code directive

ezdac committed:
Fix broken references

eorituz committed:
Restructured Documentation

eorituz committed:
Added a developer setup tutorial

eorituz committed:
Tutorials now refference the development setup for correct installation

ezdac committed:
Fix venv instructions, add venv help, fix spelling

ezdac committed:
Fix install scripts

ezdac committed:
Fix spelling, fix code directives

ezdac committed:
Add Makefile for easier development-builds

ezdac committed:
Revert "Hotfix: exlude autodoc to avoid importing of source"

This reverts commit b994f6a09e6e5fb116ec18013b9e854292511c92.

ezdac committed:
Fix wrong project path in conf

loredanacirstea commented on docs/applications/pyserver/index.rst:
```diff
@@ -0,0 +1,16 @@
+   installation
+   usage
+   cmdline
+   
```
I see there is no `py-api.rst` - why not?

eorituz replied:
You're right. It's missing for some reason.
Fixed with: https://github.com/raiden-network/microraiden/commit/ba1c77f65ffd4f179116a551ed9b3f92a85ff6ce

loredanacirstea commented on docs/applications/pyserver/index.rst:
```diff
@@ -0,0 +1,16 @@
+   installation
+   usage
+   cmdline
+   
```
Items from`Specifications` should be moved here, after command line options - those specs are proxy-server related.
```
   installation
   usage
   cmdline
   rest-api
   http-headers
   offchain sequence
   components-overview
```

loredanacirstea commented on docs/applications/pyserver/usage.rst:
```diff
@@ -0,0 +1,26 @@
+the password for the private key in the first line of the file. If it's
+not provided, the password will be prompted interactively. An Ethereum
+node RPC interface is expected to respond on http://localhost:8545.
+Alternatively, you can use `Infura
```
afaik, you could not use Infura. Did someone test and it worked?

eorituz replied:
afaik @agatsoh confirmed that the whole application worked.
@agatsoh can you confirm that Infura works in this case?

loredanacirstea commented on docs/index.rst:
```diff
@@ -6,23 +6,44 @@
-   dev-overview
-   jsclient/index
+   introduction/index
+   specifications/index
```
As I said above, these should be moved to the `pyserver` docs. I proposed an order above.
Specifications should not be first. Installation & tutorials should be the first thing that a user sees.

loredanacirstea commented on docs/index.rst:
```diff
@@ -6,23 +6,44 @@
-   jsclient/index
+   introduction/index
+   specifications/index
+   applications/index
```
Applications is a bit confusing. It makes me think of applications built on top of uRaiden. 
How about `uRaiden Components` or `uRaiden Packages`, `uRaiden Installable Components`?
This should be right after the `Introduction`.

I would have actually thought of something like this (no Applications folder, which makes it harder to understand the docs imho):
```
introduction
tutorials
pyserver
pyclient
jsclient
contract
```

ezdac replied:
we had this before and can roll it back

eorituz replied:
I like bundling pyserver, pyclient and jsclient. They kind of belong together.

loredanacirstea commented on docs/tutorials/dev-setup.rst:
```diff
@@ -0,0 +1,44 @@
+Development setup
```
This is duplicated. Installation instructions also exist in the pyserver docs. Maybe have only one place to explain installation and reference that. Just wanted you to have this in mind, doesn't need to be changed right now.

eorituz replied:
You're right, the Idea was to move all installation and setup instructions to the `Development setup` section and reference it if needed.
I'm aware of that problem and will fix it in the next version.

loredanacirstea commented on docs/tutorials/index.rst:
```diff
@@ -0,0 +1,11 @@
+
+   dev-setup
+   blockchain     
+   first-transaction
```
Consider using "First M2M microtransaction" as a name for this

eorituz replied:
I like the tutorial headline `Your first microtransaction`. It is the easiest and most straight forward tutorial and the name `Your first microtransaction` nicely encourages to try this tutorial first.

loredanacirstea commented on docs/tutorials/index.rst:
```diff
@@ -0,0 +1,11 @@
+   dev-setup
+   blockchain     
+   first-transaction
+   proxy-tutorial
```
Consider using "Paywall tutorial" as a name. "Proxy tutorial" doesn't say anything to the user.

eorituz replied:
Fixed in: https://github.com/raiden-network/microraiden/commit/1d95ca2ee50c62bfc0efc17791dcd8e5eafb424a

loredanacirstea commented on docs/applications/pyserver/index.rst:
```diff
@@ -0,0 +1,16 @@
+
+   installation
+   usage
+   cmdline
```
This is empty when I build the docs - am I missing something? 
Btw, this is quite important for both pyserver and pyclient. Same for running the examples, which should also be documented.

ezdac replied:
This will auto-document the click options, have to look into it.
What's the status on that, wasn't that not supported at the moment?

loredanacirstea replied:
> What's the status on that, wasn't that not supported at the moment?
What are you referring to?

Also, instructions for building the docs would also be nice. Maybe I missed something. I just did a `sphinx-build -b html docs build`

ezdac replied:
In the other PR #417 that's still open (and somewhat redundant, sorry), you commented:

   Docs show:
   microraiden [OPTIONS] COMMAND [ARGS]... and
   microraiden start [OPTIONS]

   Not valid commands at this point.

You probably mean that microraiden doesn't have a `scripts` entrypoint registered during setup?

The above `cmdline` docs just use a click auto-documenting plugin, so I don't know how much customization is available.
For now we can just document and explai the args passed to the example scripts manually (`--private-key` etc). This has some overlap with the click auto-documentation though.

eorituz replied:
The files `Installation.rst` and `usage.rst` are fine in my build. 
But in the file `../applications/pysercer/cmdline.rst` the section
```
.. click:: microraiden.click_helpers:main
   :prog: microraiden
   :show-nested:
```
doesn't compile correctly.

![zum beispiel](https://user-images.githubusercontent.com/3756671/36726210-4f6e7acc-1bb9-11e8-9fb4-7bf071a13eb3.png)

@ezdac could you help me out with this issue?

loredanacirstea commented on docs/api/modules.rst:
```diff
@@ -0,0 +1,7 @@
+.. toctree::
+   :maxdepth: 4
+
+   microraiden
```
Is there a current issue with the build or am I missing something? (there is no reference to the api in the docs and are not built correctly)

ezdac replied:
Fixed in #436, that was supposed to be merged before any docs merge to master

loredanacirstea replied:
So #436 is not ready yet I assume. Ok.

eorituz replied:
@ezdac Could you explain why we have the `docs/api/..` folder?
Btw it compiles fine for me. @loredanacirstea can you access the files by manually entering the URL?

loredanacirstea replied:
yes, I can access them, but I only saw headlines with no other content.

loredanacirstea commented on docs/tutorials/proxy-tutorial.rst:
```diff
@@ -2,7 +2,12 @@ Proxy tutorial
-  :maxdepth: 2 
+  :maxdepth: 2
+
+Bevor starting, please follow the
```
Before

ezdac replied:
classic german mistake

eorituz replied:
Fixed: https://github.com/raiden-network/microraiden/commit/72d2d246bc79aad505ba37c41980aeae08e9a333

loredanacirstea commented on docs/tutorials/index.rst:
```diff
@@ -0,0 +1,11 @@
+   dev-setup
+   blockchain     
+   first-transaction
+   proxy-tutorial
```
Also, not part of the current changes, but this comment in the proxy tutorial is very important and should be put somewhere much more visible:
```
Off-chain transactions are stored in a sqlite database. You should do regular backups of this file – it contains balance signatures of the client, and if you lose them, you will have no way of proving that the client is settling the channel using less funds than he has actually paid to the proxy.
```

loredanacirstea commented on docs/index.rst:
```diff
@@ -6,23 +6,44 @@
+   introduction/index
+   specifications/index
+   applications/index
    contract/index 
```
Because I cannot comment on the diagram, this should be removed: `docs/contract/diagrams/ContractClass.png`

loredanacirstea reviewed (commented):
Content wise looks good. 
I have some structure comments / questions though.

eorituz committed:
Added missing link 

Added link to py-api.rst

eorituz committed:
Added missing link 

Added link to py-api.rst

eorituz committed:
Renamed Proxy Tutorial

loredanacirstea reviewed (approved):
My comments are mostly preferences to be considered. I am approving the PR. We can discuss docs structure later if needed.

eorituz committed:
Fixed typo

ezdac committed:
Revert "Added missing link "

This reverts commit ba1c77f65ffd4f179116a551ed9b3f92a85ff6ce.
Can be reverted because of a duplicate commit, tackling the same
problem, that was waiting to get merged during the time.

ezdac committed:
"Try to fix setuptools install"

Change some requirements and loosen version restrictions

Customize theme (to raiden.network style)

Includes follwing changes, stashed together:

Add custom jquery sidebar

redesign - initial commit

redesign - bug fixes

Remove old import

color fix, mobile optimization, links fixed

show full list of contents

Delete unnessecary local build

fix sidebar, added home button, search to navbar

Fix formatting

Revert ""Try to fix setuptools install""

This reverts commit ffaa27541c8fd4679a02cd4ea75d2ce4f5cf5ce6.

ezdac committed:
"Try to fix setuptools install"

Change some requirements and loosen version restrictions

ezdac committed:
Add make clean script, add build instructions

agatsoh committed:
Adding the Raspberry pi tutorial

ezdac committed:
Include m2m diagram in tutorial

ezdac committed:
Remove duplicate search box, clean config

ezdac committed:
Add ReadTheDocs deployment config

ezdac committed:
Merge pull request #458 from ezdac/docs-develop

Add ReadTheDocs deployment config

ezdac committed:
Fix doc template build

ezdac committed:
Merge pull request #459 from ezdac/docs-develop

Fix doc template build

ezdac committed:
Fix readthedocs config syntax

ezdac committed:
Merge pull request #460 from ezdac/docs-develop

Fix readthedocs config syntax

ezdac committed:
Fix logo and sidebar depth

ezdac committed:
Merge pull request #461 from ezdac/docs-develop

Fix logo and sidebar depth

ezdac committed:
Integrate new content changes

ezdac committed:
Merge pull request #462 from ezdac/docs-develop

Integrate new content changes

agatsoh committed:
Internal document links for microraiden

agatsoh committed:
Using absolute links

ezdac committed:
Merge pull request #463 from agatsoh/internal_links_new

Internal document links for microraiden

ezdac committed:
Add cmdline args explanation, small fixes

ezdac committed:
Add legal copyright information for figures

ezdac committed:
Merge pull request #464 from ezdac/docs-develop

Add legal information to images

ezdac committed:
Change titles

ezdac committed:
Exchange raspi image file

ezdac committed:
Merge pull request #466 from ezdac/docs-develop

Change titles

ezdac committed:
Merge branch 'master' into docs-develop

krisnaer committed:
spellchecking introduction

krisnaer committed:
Fix grammar for dev-setup

- also added links for python and git.
- changed python requirement 3.5 to 3.5 or greater since the setup does not specifically require 3.5.

krisnaer committed:
Update dev-setup.rst

krisnaer committed:
Update dev-setup.rst

krisnaer committed:
Update dev-setup.rst

krisnaer committed:
Update dev-setup.rst

krisnaer committed:
Update dev-setup.rst

krisnaer committed:
spelling and formatting blockchain setup

- Blockchain setup: There are no instructions about creating a new account on geth. This is needed in the Raspberry PI tutorial.
- Deleted: Optional instructions for "geth (previously ~12) are not related to Geth. The formatting also seems off. The instructions are already provided in "Funded Ropsten account with MetaMask.

krisnaer committed:
spelling and formatting

- Added instructions to create pk_tut.txt (line 23)

krisnaer committed:
spelling and formatting

- moved the mentioning of Geth as a requirement from "running µRaiden Client and Server" (Line 50) to "Prerequisites" (Line 6) 

# Unresolved issues:
    - Is the code format Line 139-146 correct? Seems weird that there are paragraph indentations
    - Code for .JSON formatted private keys appears to be from OPs own computer (Line 101 & 141). Instructions regarding the creation and storage of these keys seems to be missing

Dominik1999 committed:
Update blockchain.rst

krisnaer committed:
Changed instructions

added imperative writing style to instructions

ezdac committed:
Merge pull request #480 from krisnaer/docs-develop

Docs develop

ezdac committed:
Merge branch 'master' into docs-develop

ezdac committed:
Revert "Fix install scripts"

This reverts commit 2f7582ac086eaf7ae1601209ae345b2e9044d549.

ezdac committed:
Ignore docs/api in git
